test/extensions/filters/network/common/fuzz/uber_filter.cc

```diff
@@ -0,0 +1,195 @@
+#include "extensions/filters/network/well_known_names.h"
+
+#include "test/extensions/filters/common/ext_authz/test_common.h"
+#include "test/test_common/utility.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+
+std::vector<absl::string_view> UberFilterFuzzer::filterNames() {
+  // This filters that have already been covered by this fuzzer.
+  // Will extend to cover other network filters one by one.
+  static ::std::vector<absl::string_view> filter_names_;
+  if (filter_names_.empty()) {
+    filter_names_ = {NetworkFilterNames::get().ExtAuthorization,
+                     NetworkFilterNames::get().LocalRateLimit,
+                     NetworkFilterNames::get().RedisProxy,
+                     NetworkFilterNames::get().ClientSslAuth,
+                     NetworkFilterNames::get().Echo,
+                     NetworkFilterNames::get().DirectResponse,
+                     NetworkFilterNames::get().DubboProxy,
+                     NetworkFilterNames::get().SniCluster};
+  }
+  return filter_names_;
+}
+
+void UberFilterFuzzer::reset() {
+  // Reset some changes made by current filter on some mock objects
+
+  // Close the connection to make sure the filter's callback is set to nullptr.
+  read_filter_callbacks_->connection_.raiseEvent(Network::ConnectionEvent::LocalClose);
+  // Clear the filter's raw pointer stored inside the connection_ and reset the connection_.
+  read_filter_callbacks_->connection_.callbacks_.clear();
+  read_filter_callbacks_->connection_.bytes_sent_callbacks_.clear();
+  read_filter_callbacks_->connection_.state_ = Network::Connection::State::Open;
+}
+void UberFilterFuzzer::perFilterSetup(const std::string& filter_name) {
+  // Set up response for ext_authz filter
+  if (filter_name == NetworkFilterNames::get().ExtAuthorization) {
+
+    async_client_factory_ = std::make_unique<Grpc::MockAsyncClientFactory>();
+    async_client_ = std::make_unique<Grpc::MockAsyncClient>();
+
+    ON_CALL(*async_client_, sendRaw(_, _, _, _, _, _))
+        .WillByDefault(testing::WithArgs<3>(Invoke([&](Grpc::RawAsyncRequestCallbacks& callbacks) {
+          Filters::Common::ExtAuthz::GrpcClientImpl* grpc_client_impl =
+              dynamic_cast<Filters::Common::ExtAuthz::GrpcClientImpl*>(&callbacks);
+          const std::string empty_body{};
+          const auto expected_headers =
+              Filters::Common::ExtAuthz::TestCommon::makeHeaderValueOption({});
+          auto check_response = Filters::Common::ExtAuthz::TestCommon::makeCheckResponse(
+              Grpc::Status::WellKnownGrpcStatus::Ok, envoy::type::v3::OK, empty_body,
+              expected_headers);
+          // Give response to the grpc_client by calling onSuccess()
+          grpc_client_impl->onSuccess(std::move(check_response), span_);
+          return async_request_.get();
+        })));
+
+    ON_CALL(*async_client_factory_, create()).WillByDefault(Invoke([&] {
+      return std::move(async_client_);
+    }));
```

Comment:
`WillByDefault` implies that this can be called multiple times.

`std::move(async_client_);` implies that the caller takes ownership of the pointer.

If this is called exactly once, use `.WillOnce()`.

Reply:
Fixed! Thanks!